Code/Framework/AzFramework/AzFramework/Input/Channels/InputChannelId.h

```diff
@@ -40,20 +41,32 @@ namespace AzFramework
         ////////////////////////////////////////////////////////////////////////////////////////////
         //! Constructor
         //! \param[in] name Name of the input channel (will be truncated if exceeds MAX_NAME_LENGTH)
-        explicit InputChannelId(const char* name = "");
-
-        ////////////////////////////////////////////////////////////////////////////////////////////
-        //! Copy constructor
-        //! \param[in] other Another instance of the class to copy from
-        InputChannelId(const InputChannelId& other);
-
-        ////////////////////////////////////////////////////////////////////////////////////////////
-        //! Copy assignment operator
-        //! \param[in] other Another instance of the class to copy from
-        InputChannelId& operator=(const InputChannelId& other);
+        explicit constexpr InputChannelId(const char* name = "")
+            : m_name(name)
+            , m_crc32(name)
+        {
+        }
 
-        ////////////////////////////////////////////////////////////////////////////////////////////
-        //! Default destructor
+        constexpr InputChannelId(const InputChannelId& other) = default;
+        constexpr InputChannelId(InputChannelId&& other) = default;
+        constexpr InputChannelId& operator=(const InputChannelId& other)
+        {
+            if (this != &other)
+            {
+                m_name = other.m_name;
+                m_crc32 = other.m_crc32;
+            }
+            return *this;
+        }
+        constexpr InputChannelId& operator=(InputChannelId&& other)
+        {
+            if (this != &other)
+            {
+                m_name = AZStd::move(other.m_name);
+                m_crc32 = AZStd::move(other.m_crc32);
+            }
```

Comment:
I think the InputChannelId being moved from should have the `m_crc32` cleared out.
The fixed_string move assignment already empties out the "copied: string.

```suggestion
            {
                m_name = AZStd::move(other.m_name);
                m_crc32 = AZStd::move(other.m_crc32);
                other.m_crc32 = 0;
            }
```